solidity/src/Router7683.sol

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.25;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { TypeCasts } from "@hyperlane-xyz/libs/TypeCasts.sol";
+
+import { BaseRouter } from "./BaseRouter.sol";
+import {
+    GaslessCrossChainOrder,
+    OnchainCrossChainOrder,
+    ResolvedCrossChainOrder,
+    Output,
+    FillInstruction,
+    IOriginSettler,
+    IDestinationSettler
+} from "./ERC7683/IERC7683.sol";
+import { OrderData, OrderEncoder } from "./libs/OrderEncoder.sol";
+
+contract Router7683 is BaseRouter, IOriginSettler, IDestinationSettler {
```

Comment:
It might be nice to create an abstract Base7683 class that does not inherit any message bridge semantics and, and then have Router7683 subclass from that?

Reply:
like this https://github.com/BootNodeDev/intents-framework/pull/1/commits/c62441d1ad61df052bc785c2de923497c0c8412e ?

Reply:
Yup, basically. I would expect some settlement logic to remain in `Base7683` (like an external `settle` method on the destination chain and an internal `settleIntent` function that would called in Router7683's handle function)

Reply:
yes, that's what I had in mind!